addons/mail/models/res_users.py

```diff
@@ -143,7 +143,7 @@ def write(self, vals):
         return write_res
 
     def action_archive(self):
-        activities_to_delete = self.env['mail.activity'].search([('user_id', 'in', self.ids)])
+        activities_to_delete = self.env['mail.activity'].sudo().search([('user_id', 'in', self.ids)])
```

Comment:
We could maybe use the "allowed_company_ids" if we want to avoid using sudo, let me know if something like this will cover all the cases for the issue, otherwise it looks good to me :)

```suggestion
        activities_to_delete = self.env['mail.activity'].with_context(allowed_company_ids=self.company_ids.ids).search([('user_id', 'in', self.ids)])
```